python/ml4ir/base/io/file_io.py

```diff
@@ -61,26 +64,17 @@ def read_df(
     Returns:
         pandas dataframe
     """
-    if infile.startswith("hdfs"):
-        # replace pydoop
-        # fp = hdfs.open(infile)
-        path_type = "HDFS"
+    if log_path:
+        log.info("Loading dataframe from path : {}".format(infile))
+
+    if infile.startswith(HDFS_PREFIX):
+        # NOTE: Move to fully spark dataframe based operations in the future
+        return spark_io.read_df(infile).toPandas()
     elif infile.endswith(".gz"):
         fp = gzip.open(os.path.expanduser(infile), "rb")
```

Comment:
I have the impression that pandas handles `gz` files out of the box.

Reply:
I used this method from our other projects. So my memory might not be completely correct - I don't think pandas can read `gz` files with the C engine. Can check though.